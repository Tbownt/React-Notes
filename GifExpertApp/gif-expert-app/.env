VITE_REACT_APP_API_KEY=U2JcARmEEoohvS73sS9SfTYBIIEitqet