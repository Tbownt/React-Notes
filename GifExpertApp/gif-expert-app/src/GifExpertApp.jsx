import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);

  const onAddCategory = (onNewCategory) => {
    if (categories.includes(onNewCategory)) return;
    setCategories([onNewCategory, ...categories]);
  };

  const onDeleteCategory = (index) => {
    setCategories(categories.filter((ele, ind) => ind !== index));
  };

  return (
    <>
      <h1 aria-label="title">Gif Expert App</h1>

      <AddCategory
        onNewCategory={(event) => onAddCategory(event)}
        categories={categories}
      />

      {categories.map((category, index) => (
        <GifGrid
          category={category}
          key={category}
          onDeleteCategory={() => onDeleteCategory(index)}
        />
      ))}
    </>
  );
};

export default GifExpertApp;
