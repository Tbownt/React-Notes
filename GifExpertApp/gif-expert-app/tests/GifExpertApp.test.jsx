import { getByText, render, screen } from "@testing-library/react";
import GifExpertApp from "../src/GifExpertApp";

describe("Pruebas en GifExpertApp", () => {
  test("Debe de matchear el snapshot ", () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });

  test("el titulo debe matchear con el ", () => {
    render(<GifExpertApp />);
    const title = screen.getByRole("heading", { level: 1 }).innerHTML;
    expect(title).toBe("Gif Expert App");
  });

  test("la aplicacion debe contener un form", () => {
    render(<GifExpertApp />);
    const form = screen.getByRole("form");
    expect(form).toBeTruthy();
  });
});
