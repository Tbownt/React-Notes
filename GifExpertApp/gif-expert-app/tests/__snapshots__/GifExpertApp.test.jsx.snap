// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe de matchear el snapshot  1`] = `
<div>
  <h1
    aria-label="title"
  >
    Gif Expert App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
